src/com/projectkorra/projectkorra/ability/WaterAbility.java

```diff
@@ -212,25 +233,96 @@ public static Block getWaterSourceBlock(final Player player, final double range,
+	}
+
+	public static boolean reduceWaterbendingSource(Player player, Block block, boolean allowPlant, boolean allowIce) {
+		return reduceWaterbendingSource(player, block, true, allowPlant, allowIce);
+	}
+
+	public static boolean reduceWaterbendingSource(Player player, Block block, boolean allowSnow, boolean allowPlant, boolean allowIce) {
+		return reduceWaterbendingSource(player, block, true, allowSnow, allowPlant, allowIce);
+	}
+
+	public static boolean reduceWaterbendingSource(Player player, Block block, boolean allowWater, boolean allowSnow, boolean allowPlant, boolean allowIce) {
+		BendingPlayer bPlayer = BendingPlayer.getBendingPlayer(player);
+		if (bPlayer == null)
+			return false;
+		allowWater = bPlayer.hasElement(Element.WATER) && allowWater;
+		allowSnow = bPlayer.hasElement(Element.WATER) && allowSnow;
+		allowPlant = bPlayer.canPlantbend() && allowPlant;
+		allowIce = bPlayer.canIcebend() && allowIce;
+		if (isWaterbendable(player, null, block) && bPlayer.hasElement(Element.WATER)) {
+			BlockData data = block.getBlockData();
+			if (allowWater && data instanceof Levelled) { // Cauldrons, just water
+				Levelled lvl = (Levelled) data;
+				if (isCauldron(block)) { // Cauldrons
+					GeneralMethods.setCauldronData(block, lvl.getLevel() - 1);
+					return true;
+				}
+				if (GeneralMethods.isAdjacentToThreeOrMoreSources(block) || lvl.getLevel() >= 8)
+					return true;
+				if (lvl.getLevel() == 7) { // lowest water lvl
+					GeneralMethods.removeBlock(block);
+					return true;
+				}
+				lvl.setLevel(Math.min(lvl.getMaximumLevel(), lvl.getLevel() + 1));
+				block.setBlockData(lvl);
+				return true;
+			} else if (allowSnow && isSnow(block)) { // layered snow
+				Snow snow = (Snow) (data instanceof Snow ? data : Material.SNOW.createBlockData());
+				int layers = (data instanceof Snow ? snow.getLayers() : snow.getMaximumLayers()) - 1;
+				if (layers >= snow.getMinimumLayers()) {
+					snow.setLayers(layers);
+					block.setBlockData(snow);
+				} else
+					GeneralMethods.removeBlock(block);
+				return true;
+			} else if (allowWater && !GeneralMethods.isAdjacentToThreeOrMoreSources(block) && data instanceof Waterlogged && ((Waterlogged) data).isWaterlogged()) { // anything waterlogged
+				Waterlogged drain = (Waterlogged) data;
+				drain.setWaterlogged(false);
+				block.setBlockData(drain);
+				return true;
+			} else if ((allowSnow && isSnow(block)) || (allowIce && isIce(block)) || (allowPlant && isPlant(block))) {
+				if (PhaseChange.getFrozenBlocksAsBlock().contains(block)) {
+					PhaseChange.thaw(block);
+					return true;
+				}
+				if (TempBlock.isTempBlock(block)) {
+					final TempBlock tb = TempBlock.get(block);
+					if (isBendableWaterTempBlock(tb)) {
+						tb.revertBlock();
+						return true;
+					}
+				} else {
+					new PlantRegrowth(player, block);
+					GeneralMethods.removeBlock(block);
+					return true;
 				}
-				return block;
 			}
 		}
```

Comment:
Any chance you can use some block statements to avoid this much indentation? Feel like it helps with readability.